### ISSUE TITLE
[Elasticsearch][Feature Request] Return the error code (e.g. 127) in Windows when APP crashes

### ISSUE THREAD

**Issue by kunisen** [elastic/elasticsearch#39621]
Currently, we have ES error code (e.g. Out of memory error : 127) documented in this doc.
https://www.elastic.co/guide/en/elasticsearch/reference/current/stopping-elasticsearch.html#_stopping_on_fatal_errors

But it seems like this error code can only be retrieved from bash script like systemctl.
https://discuss.elastic.co/t/elasticsearch-service-failing/93447
```
root@logstash01:/var/log# systemctl status elasticsearch
* elasticsearch.service - Elasticsearch
   Loaded: loaded (/usr/lib/systemd/system/elasticsearch.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Mon 2017-07-17 13:18:12 EDT; 54min ago
     Docs: http://www.elastic.co
  Process: 5308 ExecStart=/usr/share/elasticsearch/bin/elasticsearch -p ${PID_DIR}/elasticsearch.pid --quiet -E
  Process: 5304 ExecStartPre=/usr/share/elasticsearch/bin/elasticsearch-systemd-pre-exec (code=exited, status=0 Main PID: 5308 (code=exited, status=127) ＜＝ HERE
```

And this is sth un-retrievable from Windows Event log.
Only things like `Exception code: 0xc0000005` can be recognized.
https://github.com/elastic/elasticsearch/issues/15889
```
Faulting application name: elasticsearch-service-x64.exe, version: 1.0.15.0, time stamp: 0x51543b9d
Faulting module name: jvm.dll, version: 25.45.0.2, time stamp: 0x55428cbf
Exception code: 0xc0000005
Fault offset: 0x0000000000213298
Faulting process id: 0x2b7c
Faulting application start time: 0x01d14c83385f6463
Faulting application path: C:\temp\elasticsearch-2.1.1\elasticsearch-2.1.1\bin\elasticsearch-service-x64.exe
Faulting module path: c:\Program Files\Java\jdk1.8.0_45\jre\bin\server\jvm.dll
Report Id: 860f851d-b5d6-46ff-8a83-c05f54445d92
Faulting package full name:
Faulting package-relative application ID:
```

Moreover, these things cannot be detected from ES cluster log.
So for Windows users, it seems no way to know the error code and verify the specific error.
(`Exception code: 0xc0000005` is accessing violation exception, so user can make a simple guess that OOM happens here, but it is a general Windows exception but not ES related.)

In short, it is good to add this error code (e.g. Out of memory: 127) into Windows EventLog when APP crashes in Windows OS.

**Comment by rjernst** [comment]
When an OOM occurs, the Elasticsearch Java process is exiting with the expected error code. This may have been an issue when the windows service was handled through the bat script, but all the CLIs were changed to be Java based a while ago.

I'm going to close this issue for now as it hasn't had any update in quite a while. Feel free to reopen if you can provide more specifics (eg a reproduction) on when the error code is not returned correctly.
